### PR TITLE
Support case insensitive name matching to BigQuery views

### PR DESCRIPTION
**Pull request by ebyhr**
- [x] Run all BigQuery tests

### CODE REVIEW

File: plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryCaseInsensitiveMapping.java [hunk 1]
```diff
@@ -111,6 +112,40 @@ public void testNonLowerCaseTableName()
         }
     }
 
+    @Test
+    public void testNonLowerCaseViewName()
```
Comment (hashhar) [inline plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryCaseInsensitiveMapping.java:116]: As a follow-up maybe we should create an issue to test case-insensitive matching for views in all other connectors too - there is a base test class where we can add that.

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java [hunk 2]
```diff
@@ -85,13 +85,13 @@ public ConnectorSplitSource getSplits(
         log.debug("getSplits(transaction=%s, session=%s, table=%s, splitSchedulingStrategy=%s)", transaction, session, table, splitSchedulingStrategy);
         BigQueryTableHandle bigQueryTableHandle = (BigQueryTableHandle) table;
 
-        TableId tableId = bigQueryTableHandle.getRemoteTableName().toTableId();
+        TableId remoteTableId = bigQueryTableHandle.getRemoteTableName().toTableId();
```
Comment (hashhar) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java:88]: nit: if possible we can extract variable renames to it's own commit (since they are no-op changes)

Because changes in BigQuerySplitManager and ReadSessionCreator both are renames only.

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java [hunk 3]
```diff
@@ -222,16 +222,12 @@ String getProjectId()
...
+    TableId createDestinationTable(TableId remoteTableId)
     {
-        String project = viewMaterializationProject.orElse(tableId.getProject());
-        String dataset = viewMaterializationDataset.orElse(tableId.getDataset());
+        String project = viewMaterializationProject.orElse(remoteTableId.getProject());
+        String dataset = viewMaterializationDataset.orElse(remoteTableId.getDataset());
 
-        String remoteDatasetName = toRemoteDataset(project, dataset)
-                .map(RemoteDatabaseObject::getOnlyRemoteName)
-                .orElse(dataset);
```
Comment (hashhar) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java:232]: This was the actual issue - correct?

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java [hunk 4]
```diff
@@ -292,16 +288,12 @@ String selectSql(TableId table, String formattedColumns)
...
+    private String fullTableName(TableId remoteTableId)
     {
-        String remoteSchemaName = toRemoteDataset(tableId.getProject(), tableId.getDataset())
-                .map(RemoteDatabaseObject::getOnlyRemoteName)
-                .orElse(tableId.getDataset());
-        String remoteTableName = toRemoteTable(tableId.getProject(), remoteSchemaName, tableId.getTable())
-                .map(RemoteDatabaseObject::getOnlyRemoteName)
-                .orElse(tableId.getTable());
-        tableId = TableId.of(tableId.getProject(), remoteSchemaName, remoteTableName);
-        return format("%s.%s.%s", tableId.getProject(), tableId.getDataset(), tableId.getTable());
```
Comment (hashhar) [inline plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java:304]: This was the issue - correct? We already had the remote name but were trying to translate it anyway.

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java (hunk 2 above)
Comment (hashhar) [reply plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQuerySplitManager.java:88]: I now think it'd be much easier to figure out the actual bugfix after extracting a commit.

**Review by hashhar — COMMENTED** [review]
Looks good. Some comments.

A question:

Do we have any remaining methods in BigQueryClient that try to do call the `#toRemoteTable` or `#toRemoteDataset`. Ideally as long as BigQueryMetadata prepares the table handle with remote names no class except BigQueryMetadata should need to call those methods.

File: plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryCaseInsensitiveMapping.java (hunk 1 above)
Comment (hashhar) [reply plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryCaseInsensitiveMapping.java:116]: https://github.com/trinodb/trino/issues/9165

File: plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java (hunk 4 above)
Comment (ebyhr) [reply plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java:304]: Correct, here and the above places were the issue.

**Comment by ebyhr** [comment]
> Do we have any remaining methods in BigQueryClient that try to do call the #toRemoteTable or #toRemoteDataset

There is no usage in BigQueryClient after this change.

**Comment by ebyhr** [comment]
@hashhar Updated and confirmed test result again.

**Review by hashhar — APPROVED** [review]
LGTM - looks good.

It's also so nice that all the remote name logic is now present only in BigQueryMetadata.
